### PR TITLE
Refactor expected links test

### PR DESCRIPTION
**Pull request by pikinier20**
@nicolasstucki 
@smarter

### CODE REVIEW

File: project/Build.scala [hunk 1]
```diff
@@ -1376,7 +1376,12 @@ object Build {
         generateDocumentation(Testcases)
       }.value,
 
-      generateReferenceDocumentation := Def.taskDyn {
+      generateReferenceDocumentation := Def.inputTaskDyn {
+        val shouldRegenerateExpectedLinks = spaceDelimited("[--no-regenerate-expected-links]").parsed match {
+          case "--no-regenerate-expected-links" :: rest => false
+          case _ => true
+        }
```
Comment (julienrf) [inline project/Build.scala:1383]: The idea is to _not_ pass the option `--no-regenerate-expected-links` in another workflow? Or is this used by some workflow already?

I suggest maybe turning things around and naming the option `--regenerate-expected-links`. Also, I suggest using the `literal` constructor:

```suggestion
        val shouldRegenerateExpectedLinks = literal("--regenerate-expected-links").?.parsed.isDefined
```

Comment (julienrf) [reply project/Build.scala:1383]: Ah true. OK. Do you mind still following my suggestion of using `literal` instead of `spaceDelimited`?

Comment (pikinier20) [reply project/Build.scala:1383]: Of course, I'm not so familiar with sbt parsers so I need to trust you :)

Comment (pikinier20) [reply project/Build.scala]: I applied the change

**Review by julienrf — APPROVED** [review]

**Pull request merged**